src/test/java/uk/nhs/ctp/pageobject/ManageEmsSuppliersPage.java

```diff
@@ -0,0 +1,101 @@
+package uk.nhs.ctp.pageobject;
+
+import static org.openqa.selenium.support.ui.ExpectedConditions.textToBePresentInElementValue;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOf;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfAllElements;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.checkerframework.checker.nullness.compatqual.NullableDecl;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import uk.nhs.ctp.model.EmsSupplier;
+import uk.nhs.ctp.model.TableRow;
+import uk.nhs.ctp.util.SeleniumUtil;
+
+public class ManageEmsSuppliersPage extends PageObject {
+
+  public ManageEmsSuppliersPage(WebDriver driver) {
+    super(driver);
+  }
+
+  @FindBy(tagName = "app-ems-supplier")
+  private WebElement component;
+
+  @FindBy(xpath = "//span[contains(text(), \"Create New Supplier\")]")
+  private WebElement createButton;
+
+  @FindBy(css = "input.emsName")
+  private WebElement nameTextBox;
+
+  @FindBy(css = "input.emsBaseUrl")
+  private WebElement baseUrlTextBox;
+
+  @FindBy(css = "input.emsAuthToken")
+  private WebElement authTokenTextBox;
+
+  @FindBy(css = "button.saveEms")
+  private WebElement saveButton;
+
+  @FindBy(css = "tr.emsSupplier")
+  private List<WebElement> supplierRows;
+
+  @Override
+  public boolean onPage() {
+    return wait.until(visibilityOf(component)).isDisplayed();
+  }
+
+  public void create(EmsSupplier newSupplier) {
+    wait.until(visibilityOf(createButton)).click();
+    wait.until(visibilityOfAllElements(nameTextBox, baseUrlTextBox));
+
+    SeleniumUtil.setValue(nameTextBox, newSupplier.getName(), wait);
+    SeleniumUtil.setValue(baseUrlTextBox, newSupplier.getBaseUrl(), wait);
+    SeleniumUtil.setValue(authTokenTextBox, newSupplier.getAuthToken(), wait);
+
+    saveButton.click();
+  }
+
+  public void suppliersListNonEmpty() {
+    wait.until(visibilityOfAllElements(supplierRows));
+  }
+
+  public List<EmsSupplier> supplierList() {
+    return supplierRows.stream()
+        .map(TableRow::new)
+        .map(EmsSupplier::from)
+        .collect(Collectors.toList());
+  }
+
+  public Optional<WebElement> supplierRow(EmsSupplier emsSupplier) {
```

Comment:
super minor, but just so that we don't interact with webelements in steps, this should be private